src/convex/ConvexRewardPoolController.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import {IController} from "../core/IController.sol";
+
+interface ICurveGauge {
+    function lp_token() external view returns (address);
+}
+
+interface IRewardPool {
+    function curveGauge() external view returns (address);
+    function rewardLength() external view returns (uint256);
+    function rewards(uint index) external view returns (RewardType memory);
+}
+
+struct RewardType {
+    address reward_token;
+    uint128 reward_integral;
+    uint128 reward_remaining;
+}
+
+contract ConvexRewardPoolController is IController {
+
+    /// @notice withdraw(uint256, bool)
+    bytes4 WITHDRAW = 0x38d07436;
+
+    /// @notice getReward(address)
+    bytes4 GET_REWARD = 0xc00007b0;
+
+    function canCall(address target, bool, bytes calldata data)
+    external
+    view
+    returns (bool, address[] memory, address[] memory) {
+        bytes4 sig = bytes4(data);
+
+        if (sig == WITHDRAW) return canWithdraw(target);
+        if (sig == GET_REWARD) return canGetReward(target);
+
+        return (false, new address[](0), new address[](0));
+    }
+
+    function canWithdraw(address rewardPool) 
+    internal
+    view
+    returns (bool, address[] memory, address[] memory) {
+        address[] memory tokensIn = new address[](1);
+        tokensIn[0] = ICurveGauge(IRewardPool(rewardPool).curveGauge()).lp_token();
```

Comment:
If claim is set to true, the controller won't account for the reward tokens coming into the account